sunpy/net/scraper.py

```diff
@@ -350,23 +283,28 @@ def _ftpfileslist(self, timerange):
 
     def _localfilelist(self, timerange):
         pattern = self.pattern
```

Comment:
Is there a reason to define this here?

Reply:
yes, to hold the original value while we change the class variable `pattern` for the body of the function and then put back on in the end